Extension/src/LanguageServer/settings.ts

```diff
@@ -560,13 +560,13 @@ export class CppSettings extends Settings {
 
     public get dimInactiveRegions(): boolean {
         return super.Section.get<boolean>("dimInactiveRegions") === true
-            && super.Section.get<string>("intelliSenseEngine") === "Default"
+            && super.Section.get<string>("intelliSenseEngine")?.toLowerCase() === "default"
             && vscode.workspace.getConfiguration("workbench").get<string>("colorTheme") !== "Default High Contrast";
     }
 
     public toggleSetting(name: string, value1: string, value2: string): void {
         const value: string | undefined = super.Section.get<string>(name);
-        super.Section.update(name, value === value1 ? value2 : value1, getTarget());
+        super.Section.update(name, value === value1.toLowerCase() ? value2.toLowerCase() : value1.toLowerCase(), getTarget());
```

Comment:
I think this should be:

```
super.Section.update(name, value.toLowerCase() === value1.toLowerCase() ? value2 : value1, getTarget());
```
I assume the values should be lowercased for comparison but set to the values provided in the call to `toggleSettings`.